### PR TITLE
Server Invite Developers feature.

### PR DESCRIPTION
**Pull request by JackCrispy**
Lets the devs get an invite to any server Switchblade is in.
Usage: !serverinvite GUILD_ID

### CODE REVIEW

**Comment by dpaiv0** [comment]
• fix Standard issues (npm run fix)
• missing commandUsage (```<guild ID>```)

**Comment by pedrofracassi** [comment]
I don't really think we should do this. Most of the times, users don't really like random people joining their servers - which might be private.

**Comment by pedrofracassi** [comment]
- Why are you using StringParameter instead of GuildParameter?
- We have refactored the way command constructors work. Please check out the new code and make changes so your code fits the new standard.

File: src/commands/developers/serverinvite.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    }
+    
+    async run({ channel, message }, expr) {
+        // Add permission handling here
+        const guildID = expr; // Add args handling here
+        if (!(guildID && message.client.guilds.has(guildID))) return message.channel.send(`Guild not found`);
+        message.client.guilds.get(guildID).channels.filter((channel) => channel.permissionsFor(message.guild.me).has(1) && channel.type !== 'category').first().createInvite({ maxAge: 0 }).then((invite) => {
+        message.channel.send(`Here's the invite: ${invite.url}`);
+        }).catch((error) => {
+                 return message.channel.send('I ran into an error: ' + error);
```
Comment (pedrofracassi) [inline src/commands/developers/serverinvite.js]: We don't do that here. Use `throw` and a `CommandError` instead.

File: src/commands/developers/serverinvite.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        this.requirements = new CommandRequirements(this, { devOnly: true });
+        this.parameters = new CommandParameters(this, new StringParameter({ full: true, missingError: 'errors:missingParameters', showUsage: false }));
+    }
+    
+    async run({ channel, message }, expr) {
+        // Add permission handling here
+        const guildID = expr; // Add args handling here
+        if (!(guildID && message.client.guilds.has(guildID))) return message.channel.send(`Guild not found`);
+        message.client.guilds.get(guildID).channels.filter((channel) => channel.permissionsFor(message.guild.me).has(1) && channel.type !== 'category').first().createInvite({ maxAge: 0 }).then((invite) => {
+        message.channel.send(`Here's the invite: ${invite.url}`);
```
Comment (pedrofracassi) [inline src/commands/developers/serverinvite.js]: Please use a SwitchbladeEmbed

File: src/commands/developers/serverinvite.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        this.hidden = true;
+        
+        this.requirements = new CommandRequirements(this, { devOnly: true });
+        this.parameters = new CommandParameters(this, new StringParameter({ full: true, missingError: 'errors:missingParameters', showUsage: false }));
+    }
+    
+    async run({ channel, message }, expr) {
+        // Add permission handling here
+        const guildID = expr; // Add args handling here
+        if (!(guildID && message.client.guilds.has(guildID))) return message.channel.send(`Guild not found`);
```
Comment (pedrofracassi) [inline src/commands/developers/serverinvite.js]: throw a CommandError insteae

File: src/commands/developers/serverinvite.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+module.exports = class Serverinvite extends Command {
+    constructor(client) {
+        super(client);
+        this.name = 'serverinvite';
+        this.aliases = ['sinvite'];
+        this.category = 'developers';
+        this.hidden = true;
+        
+        this.requirements = new CommandRequirements(this, { devOnly: true });
+        this.parameters = new CommandParameters(this, new StringParameter({ full: true, missingError: 'errors:missingParameters', showUsage: false }));
```
Comment (pedrofracassi) [inline src/commands/developers/serverinvite.js]: Use a GuildParameter instead of a String one

**Pull request closed without merging**